pyccel/parser/semantic.py

```diff
@@ -2883,6 +2903,9 @@ def _insert_obj(location, target, obj):
                             _insert_obj('functions', name, atom)
             else:
                 _insert_obj('variables', source_target, imports)
+            key = expr.source if isinstance(expr.source, AsName) else source
+            _insert_obj('imports', key, Import(source, expr.target, True))
```

Comment:
Are you sure you can call `_insert_obj` twice in a row for the same object given that this function raises errors if an object exists already? Could you not use your `insert_import` function here?

Reply:
that's what I'm trying to do I need to tweak  the function a little